File: backend/internal/application/ports/event.go
package ports

import (
	"context"
	"time"

	"github.com/gdg-eskisehir/events/backend/internal/domain"
)

// EventListFilter controls listing events for public vs admin surfaces.
type EventListFilter struct {
	PublishedOnly bool
	Status        *domain.EventStatus
}

type EventRepository interface {
	GetByID(ctx context.Context, eventID string) (*domain.Event, error)
	GetRegistrationCount(ctx context.Context, eventID string) (int, error)
	List(ctx context.Context, filter EventListFilter) ([]*domain.Event, error)
	Insert(ctx context.Context, e *domain.Event) error
	Update(
		ctx context.Context,
		id string,
		title, description *string,
		capacity *int,
		startsAt, endsAt *time.Time,
	) error
	SetStatus(ctx context.Context, id string, status domain.EventStatus) error
}

File: apps/mobile/lib/features/events/domain/entities/event.dart
import 'package:equatable/equatable.dart';
import 'package:gdg_events/features/events/domain/event_status.dart';

class Event extends Equatable {
  const Event({
    required this.id,
    required this.title,
    required this.description,
    required this.status,
    required this.capacity,
    required this.startsAt,
    required this.endsAt,
  });

  final String id;
  final String title;
  final String? description;
  final EventStatus status;
  final int capacity;
  final DateTime startsAt;
  final DateTime endsAt;

  bool get isRegisterable => status == EventStatus.published;

  @override
  List<Object?> get props =>
      [id, title, description, status, capacity, startsAt, endsAt];
}
